src/impl_methods.rs

```diff
@@ -1968,6 +1968,46 @@ where
         self.zip_mut_with_by_rows(rhs, f);
     }
 
+    /// Traverse two arrays in unspecified order, in lock step,
+    /// calling the closure `f` on each element pair, and put
+    /// the result into the corresponding element of self.
+    pub fn zip_mut_from_pair<B, C, S1, S2, F>(&mut self, lhs: &ArrayBase<S1, D>, rhs: &ArrayBase<S2, D>, f: F, )
+    where
+        S: DataMut,
+        S1: Data<Elem = B>,
+        S2: Data<Elem = C>,
+        F: Fn(&B, &C) -> A,
+    {
```

Comment:
Why not just use `Zip` in this method? `Zip::map_assign_into` should do it (assign into self)

Reply:
Right, we can use Zip. This function does not need to be added.